src/App.tsx

```diff
@@ -15,48 +16,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+
+function getPreparedGoods(
+  goods: string[],
+  sortByType: string,
+  isReverse: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortByType) {
+    return preparedGoods.sort((good1: string, good2: string) => {
+      switch (sortByType) {
+        case SORT_BY_ALPHABET:
+          return isReverse
+            ? good2.localeCompare(good1)
+            : good1.localeCompare(good2);
+        case SORT_BY_LENGTH:
+          if (good1.length === good2.length) {
+            return isReverse
+              ? good2.localeCompare(good1)
+              : good1.localeCompare(good2);
+          }
+
+          return isReverse
+            ? good2.length - good1.length
+            : good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  const goods = [...goodsFromServer];
+
+  const preparedGoods = getPreparedGoods(goods, sortBy, isReverse);
+
+  const handleIsReverse = () => setIsReverse(!isReverse);
```

Comment:
When you plan to use your previous state to change next state - it's always preferable to write it that way: 

`const handleIsReverse = () => setIsReverse(prevReversed => !prevReversed);`

This may save you from some bugs due to useState hook nature.